src/seedu/addressbook/data/person/Address.java

```diff
@@ -35,10 +35,15 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         String[] addressParts = address.split(", ");
-        this.block = new Block(addressParts[AddressCSVIndices.BLOCK]);
-        this.street = new Street(addressParts.length > 1 ? addressParts[AddressCSVIndices.STREET] : "");
-        this.unit = new Unit(addressParts.length > 2 ? addressParts[AddressCSVIndices.UNIT] : "");
-        this.postalCode = new PostalCode(addressParts.length > 3 ? addressParts[AddressCSVIndices.POSTAL_CODE] : "");
+        try {
```

Comment:
Actually the regex "^[^,]*(, [^,]*){3}$" checks for exactly 3 commas so the checking of length is made redundant. My test cases have been updated to check for an error if address is not "<something>, [something], [something], [something]"